lightning/src/routing/network_graph.rs

```diff
@@ -1040,19 +1066,87 @@ impl NetworkGraph {
 		}
 	}
 
+	#[cfg(feature = "std")]
+	/// Removes information about channels that we haven't heard any updates about in some time.
+	/// This can be used regularly to prune the network graph of channels that likely no longer
+	/// exist.
+	///
+	/// While there is no formal requirement that nodes regularly re-broadcast their channel
+	/// updates every two weeks, the non-normative section of BOLT 7 currently suggests that
+	/// pruning occur for updates which are at least two weeks old, which we implement here.
+	///
+	/// Note that for users of the `lightning-background-processor` crate this method may be
+	/// automatically called regularly for you.
+	///
+	/// This method is only available with the `std` feature. See
+	/// [`NetworkGraph::remove_stale_channels_with_time`] for `no-std` use.
+	pub fn remove_stale_channels(&self) {
+		let time = SystemTime::now().duration_since(UNIX_EPOCH).expect("Time must be > 1970").as_secs();
+		self.remove_stale_channels_with_time(time);
+	}
+
+	/// Removes information about channels that we haven't heard any updates about in some time.
+	/// This can be used regularly to prune the network graph of channels that likely no longer
+	/// exist.
+	///
+	/// While there is no formal requirement that nodes regularly re-broadcast their channel
+	/// updates every two weeks, the non-normative section of BOLT 7 currently suggests that
+	/// pruning occur for updates which are at least two weeks old, which we implement here.
+	///
+	/// This function takes the current unix time as an argument. For users with the `std` feature
+	/// enabled, [`NetworkGraph::remove_stale_channels`] may be preferable.
+	pub fn remove_stale_channels_with_time(&self, current_time_unix: u64) {
+		let mut channels = self.channels.write().unwrap();
```

Comment:
Wonder if we should document the locking order requirements to make sure this never deadlocks with writing the network graph to disk

Reply:
Hmm? Not sure what you're referring to here. The lockorder between channels/nodes is documented, and the locks are otherwise internal, so shouldn't deadlock?

Reply:
> The lockorder between channels/nodes is documented

Missed that, thanks